docs/build/tutorials/smart-contract-development/account-abstraction/custom-aa-tutorial.md

```diff
@@ -842,6 +842,12 @@ Make sure you deposit funds on zkSync Era using [one of the available bridges](h
 
 1. In the `deploy` folder, create the file `deploy-factory.ts` and copy/paste the following code, replacing `<WALLET_PRIVATE_KET>` with your private key.
 
+::: warning
+Remember to use a private key that only has testnet funds if you are going to paste your private key in plaintext. It is recommended to [never have a private key associated with real funds in plaintext.](https://github.com/Cyfrin/foundry-full-course-f23/discussions/5).
+
+If you wish to use hardhat with real funds, it's recommended to encrypt your key so it's not accidentally exposed.
```

Comment:
Do you have a link to an example we can link to for encrypting your Private Key for use with something like Hardhat?

Also, as a general note, the upcoming new docs will lean more on tools like Foundry's `cast wallet import`to manage keys in a private encrypted keystore automatically for devs.

Reply:
For Hardhat, you have to roll your own encryption script. They recommend using their variables setting, but I don't like it because the keys are still in plaintext and just moved elsewhere. 

[I have used this example in the past to encrypt and decrypt your own keys.](https://github.com/PatrickAlphaC/ethers-simple-storage-fcc/blob/main/encryptKey.js) The project has an example of encrypting and decrypting the keys. 

`cast wallet import` is the best way to do key management!! That's great! 

I'll add this example to the warning.